tests/unit/rest.py

```diff
@@ -231,74 +229,165 @@ def test_none_return(self):
+    # 4. Arguments in body and not in the URL.
+    @rest.rest_call('POST', '/body/args', Schema({
+        'arg1': basestring,
+        'arg2': basestring,
+    }))
+    def body_args(arg1, arg2):
+        return json.dumps([arg1, arg2])
 
-        `data` should be a string -- the server will expect valid json, but
-        we want to write test cases with invalid input as well.
-        """
-        resp = self.client.post('/give-me-an-e', data=data)
-        return resp
-
-    def test_ok(self):
-        assert not _is_error(self._do_request(json.dumps({'foo': 'alice',
-                                                          'bar': 'bob'})),
-                             rest.ValidationError)
-
-    def test_bad_json(self):
-        # Illegal JSON gets caught by flask itself, before we get to look at
-        # it. As such, the resulting error isn't an instance of APIError, so
-        # we can't use  `_is_error` here.
-        resp = self._do_request('xploit')
-        assert resp.status_code == 400
-
-    def test_missing_bar(self):
-        assert _is_error(self._do_request(json.dumps({'foo': 'hello'})),
-                         rest.ValidationError)
-
-    def test_extra_baz(self):
-        assert _is_error(self._do_request(json.dumps({'foo': 'alice',
-                                                      'bar': 'bob',
-                                                      'baz': 'eve'})),
-                         rest.ValidationError)
-
-    def test_mixed_args_ok(self):
-        """Test a call that has arguments in both the url and body."""
-        resp = self.client.put('/mixed/args/foo',
-                               data=json.dumps({'arg2': 'bar'}))
-        assert resp.status_code == 200
-        assert json.loads(resp.get_data()) == ['foo', 'bar']
+    # Let's also make sure we're testing something with a schema that isn't
+    # just basestring:
+    @rest.rest_call('PUT', '/non-string-schema', schema=Schema({
+        "the_value": int,
+    }))
+    def non_string_schema(the_value):
+        return json.dumps(the_value)
+
+
+def _do_request(client, method, path, data):
+    """Make a request to the endpoint with `data` in the body.
 
-    def test_custom_schema(self):
-        assert _is_error(self.client.put('/custom-schema', data=json.dumps({
-            'the_value': 'Not an integer!',
-        })), rest.ValidationError)
+    `client` is a flask test client
+
+    `method` is the request method
+
+    `path` is the path of the request
+
+    `data` should be a string -- the server will expect valid json, but
+    we want to write test cases with invalid input as well.
+    """
+    client.post('/give-me-an-e', data=data)
+    # Annoyingly, the client doesn't give us a way to supply an arbitrary
+    # HTTP method, so we hack around it by using getattr to get the right
+    # function:
+    return getattr(client, method.lower())(path, data=data)
+
+
+@pytest.mark.parametrize('args', [
```

Comment:
I really like this parameterize!!

Would you mind also adding test cases to cover:
- [ ] duplicated args in the URL and body
- [ ] An empty body for `body_args`